libp2p/muxers/mplex/lpchannel.nim

```diff
@@ -93,12 +93,47 @@ proc resetMessage(s: LPChannel) {.async.} =
   await s.conn.writeMsg(s.id, s.resetCode)
 
 proc resetByRemote*(s: LPChannel) {.async.} =
-  await allFutures(s.close(), s.closedByRemote())
+  # Immediately block futher calls
   s.isReset = true
-  await s.cleanUp()
+
+  # start and await async teardown
+  let
+    futs = await allFinished(
+      s.close(),
+      s.closedByRemote(),
+      s.cleanUp()
+    )
+
+  for res in futs:
+    if res.failed:
+      let exc = res.readError()
+      if exc of LPStreamEOFError:
+        # EOF are likely to happen here
+        trace "Got a non fatal error", failure = exc.name, msg = exc.msg
+      else:
+        # We still don't abort but warn
+        warn "Something went wrong during LPChennel.resetByRemote",
+         failure = exc.name, msg = exc.msg
+
 
 proc reset*(s: LPChannel) {.async.} =
-  await allFutures(s.resetMessage(), s.resetByRemote())
+  let
+    futs = await allFinished(
+      s.resetMessage(),
+      s.resetByRemote()
+    )
+
+  # NOTICE WE DO NOT RETHROW
```

Comment:
this might seem very dramatic right now because we've had long discussions about it, but at the end of the day, it's not that unusual to consciously decide not to do anything with an error ;)

Reply:
well the comment is a insurance to avoid a future conversation :smile